agreement/models/agreement.py

```diff
@@ -342,11 +342,15 @@ def onchange_sub_object_id(self):
 
     @api.onchange('sub_model_object_field_id', 'default_value')
```

Comment:
I think "field_id" should be added in line 343:
 @api.onchange('sub_model_object_field_id', 'default_value', 'field_id')

Reply:
it works but still when editing, the "sub_object_id" field is not reset and shows the previous value.

Reply:
Got it. Should be fixed now.